src/components/forms/profile/update-user-address-form.tsx

```diff
@@ -0,0 +1,136 @@
+import React from "react";
+import { v4 as uuidv4 } from "uuid";
+
+import { InputText } from "../../input/input-text";
+// import { WrongInputError } from "../input/wrong-input-error";
+import { Form } from "../form";
+import { Address } from "@src/@types/profile";
+import { Button, ButtonVariant } from "@src/components/buttons/buttons";
+import { FakeButton } from "@src/components/buttons/fake-button";
+import { NeutralLink } from "@src/components/neutral-link/neutral-link";
+
+const UpdateUserAddressForm: React.FC<{
+  userAddress?: Address;
+}> = ({ userAddress }) => {
+  const [formID, setFormID] = React.useState<string>(uuidv4());
+
+  const [updatedAddress, setUpdatedAddress] = React.useState<Address>({
+    id: "",
+    sub: "",
+    street_address: "string",
+    locality: "string",
+    region: "string",
+    postal_code: "string",
+    country: "string",
+    kind: "string",
+    created_at: "string",
+    updated_at: "string",
+    street_address_2: "string",
+    primary: false,
+  });
+
+  React.useEffect(() => {
+    if (userAddress) {
+      setUpdatedAddress(userAddress);
+    }
+  }, [userAddress]);
+
+  return (
+    <>
+      <Form
+        formID={formID}
+        onSubmit={async () => {
```

Comment:
```suggestion
        onSubmit={() => {
```

Reply:
Well, this one is an "issue" with the `Form` component https://github.com/fewlinesco/connect-account/blob/main/src/components/forms/form.tsx. We didn't want to handle sync and async callbacks, so we typed it as async, hence the useless async keyword. I don't really see this as an issue, but if you do, let me know, and we'll make a task to handle both.

Reply:
Yeah, the overhead is negligible, but the "wrongly interpreted" is what bothers me in this, that's why I would be ok to make the changes in the Form component at some points.